lib/ide/src/view/view_layout.rs

```diff
@@ -0,0 +1,159 @@
+//! This module contains implementation of ViewLayout with a single TextEditor temporarily
+//! occupying half bottom of the screen temporarily as the default layout.
+
+use wasm_bindgen::prelude::*;
+
+use super::text_editor::TextEditor;
+use super::ui_component::UiComponent;
+
+use basegl::system::web::*;
+use basegl::display::world::World;
+
+use nalgebra::zero;
+use nalgebra::Vector2;
+use std::rc::Rc;
+use std::cell::RefCell;
+use js_sys::Function;
+use web_sys::KeyboardEvent;
+use wasm_bindgen::JsCast;
+use crate::view::ui_component::Padding;
+
+
+//TODO: ViewMode is a temporary enumeration, it will be replaced by proper Panel impl.
+// ================
+// === ViewMode ===
+// ================
+
+/// Defines the element's view mode. It can fully occupy the screen or only half of it.
+#[derive(Clone,Copy,Debug)]
+enum ViewMode {
+    Full,
+    Half
+}
+
+
+
+// ======================
+// === ViewLayoutData ===
+// ======================
+
+#[derive(Debug)]
+struct ViewLayoutData {
+    text_editor      : TextEditor,
+    keyboard_closure : Option<Closure<dyn FnMut(KeyboardEvent)>>,
+    view_mode        : ViewMode,
+    dimensions       : Vector2<f32>
+}
+
+impl Drop for ViewLayoutData {
+    fn drop(&mut self) {
+        if let Some(keyboard_closure) = self.keyboard_closure.as_ref() {
+            let body = document().unwrap().body().unwrap();
+            let callback : &Function = keyboard_closure.as_ref().unchecked_ref();
+            body.remove_event_listener_with_callback("keydown", callback).ok();
```

Comment:
This should be done more safe. You are deteaching the callback from "keydown" because you have registered it there. What happens if I will change the code in 2 months and attach it to something else but I will forget updating this drop? Memory leak.